lib/matplotlib/rcsetup.py

```diff
@@ -1180,6 +1180,7 @@ def _validate_linestyle(ls):
 
     'axes.titlesize':        ['large', validate_fontsize],  # fontsize of the
                                                             # axes title
+    'axes.titlelocation':    ['center', validate_string],  # alignment of axes title
```

Comment:
Maybe the validation should already make sure only valid locations are passed? E.g. adding a 

    validate_titlelocation = ValidateInStrings('axes.titlelocation', ['left', 'center', 'right'])

and using it as

    'axes.titlelocation':    ['center', validate_titlelocation],  # alignment of axes title